core/thread/src/TReentrantRWLock.cxx

```diff
@@ -322,6 +322,12 @@ TReentrantRWLock<MutexT, RecurseCountsT>::Rewind(const State &earlierState) {
    if (pStateDelta->fDeltaReadersCount != 0) {
       // Claim a recurse-state +1 to be able to call Unlock() below.
       *typedState.fReadersCountLoc = typedState.fReadersCount + 1;
+      // ReadersCount and Readers can be read/updated before actually holding the R/W lock.
+      // (See WriteLock and ReadLock methods) but fReaders is an atomic, so that should be fine
+      // but that's weird, that does not account to other change in fReaders during between
+      // the snapshot and the rewind ... humm unless the lock held is a WriteLock
```

Comment:
Should / can we `assert` that the write lock is taken?

Reply:
I added some error message.  Let me know if you agree.